packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/plugin.ts

```diff
@@ -56,23 +60,118 @@ export class WebpackAngularTypesPlugin {
 				smallTsProject.resolveSourceFileDependencies();
 
 				for (const { path, module } of modulesToProcess) {
-					const { classesInformation, interfacesInformation } = generateTypeInformation(
+					const {
+						classesInformation,
+						interfacesInformation,
+						functionsInformation,
+						constantsInformation,
+					} = generateTypeInformation(
 						path,
 						smallTsProject,
 						this.options.excludeProperties,
 					);
-					for (const ci of classesInformation) {
-						this.addClassCodeDocDependency(ci, module);
+					for (const classInformation of classesInformation) {
+						this.addClassCodeDocDependency(classInformation, module);
 					}
-					for (const ii of interfacesInformation) {
-						this.addInterfaceCodeDocDependency(ii, module);
+					for (const interfaceInformation of interfacesInformation) {
+						this.addInterfaceCodeDocDependency(interfaceInformation, module);
+					}
+
+					const groupedExportsInformation = this.groupExportInformation(
+						functionsInformation,
+						constantsInformation,
+					);
+
+					for (const groupedExportInformation of groupedExportsInformation) {
+						this.addGroupedExportsCodeDocDependency(groupedExportInformation, module);
 					}
 				}
 				this.moduleQueue = [];
 			});
 		});
 	}
 
+	/**
+	 * Functions and constants can be annotated with "@group-docs" to collect them into a
+	 * single ArgsTable. Returns information grouped by the "groupBy" property.
+	 *
+	 * @param functionsInformation list of all functions
+	 * @param constantsInformation list of all constants
+	 */
+	private groupExportInformation(
+		functionsInformation: FunctionInformation[],
+		constantsInformation: ConstantInformation[],
+	): GroupedExportInformation[] {
+		const exportsInformationMap = new Map<string, ExportsInformation>();
+
+		for (const functionInformation of functionsInformation) {
+			const name = functionInformation.name;
+
+			exportsInformationMap.set(name, {
+				functionsInformation: [functionInformation],
+				constantsInformation: [],
+			});
+
+			const groupBys = functionInformation.groupBy;
+
+			for (const groupBy of groupBys) {
+				if (groupBy) {
+					const groupByEntry = exportsInformationMap.get(groupBy);
+					if (groupByEntry) {
+						if (groupByEntry.functionsInformation.length) {
+							groupByEntry.functionsInformation.push(functionInformation);
+						} else {
+							groupByEntry.functionsInformation = [functionInformation];
+						}
+					} else {
+						exportsInformationMap.set(groupBy, {
+							functionsInformation: [functionInformation],
+							constantsInformation: [],
+						});
+					}
+				}
+			}
+		}
```

Comment:
This seems a bit weird. In 110, you map the function name to the information.
But a few lines below you retrieve and set data on this map based on the group name.

Is the first `.set` on the map in there accidentally?

Reply:
No, I wanted to make it accessible by both its name (e.g. "myFancyFn") and the groupBy params (e.g. "Group1", "Group2"). Do you think this makes sense?

Reply:
But the value for a key based on the function/constant name only contains one element, right? What is the point of it? And for what do you need it?

You only need to access the functions/constants group-wise in the end

Reply:
Also see the other thread that I opened on practially the same code after the most recent commit.

Reply:
> You only need to access the functions/constants group-wise in the end

Is there no use case for just displaying information on a single function? If you want I can change that, but that would also mean that you always have to set the "@group-docs" annotation. And if its mandatory we could merge it with "@include-docs":

```
* @include-docs Group1, Group2
```

Reply:
Oh I completely overlooked this usecase 💡 . Of course it should be possible to emit docs without grouping. I just overlooked that this would not work without registering the constant/function by name 🤦 

But one more thing: Currently it would not be possible to register functions with the same name (but from different modules), right?